.github/workflows/pre-post-release.yml

```diff
@@ -0,0 +1,97 @@
+---
+name: Pre/Post Release
+
+on:
+  workflow_call:
+    ref:
+      description: 'Branch or tag ref to run the workflow on'
+      required: true
+      default: 'main'
+    version:
+      description: 'The version to release (e.g. 1.2.3). This workflow will automatically perform the required version bumps'
+      required: true
+    phase:
+      description: 'Pre or post release phase'
+      type: choice
+      options:
+        - pre
+        - post
+      required: true
+
+env:
+  RELEASE_VERSION: ${{ inputs.version }}
+  BRANCH_NAME: ${{ inputs.phase }}-release-v${{ inputs.version }}
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}
+
+jobs:
+  validate-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Validate tag does not exist on current commit
```

Comment:
[question] does it validates that the tag does not yet exist on current commit or that the tag hasn't been created on the whole repository ?